qiskit_ibm/runtime/runtime_job.py

```diff
@@ -203,15 +199,17 @@ def wait_for_final_state(
             JobTimeoutError: If the job does not reach a final state before the
                 specified timeout.
         """
+        if not self._is_streaming():
+            self._ws_client_future = self._executor.submit(self._start_websocket_client)
+        stream_status = self._is_streaming()
         start_time = time.time()
-        status = self.status()
-        while status not in JOB_FINAL_STATES:
+        while stream_status:
             elapsed_time = time.time() - start_time
             if timeout is not None and elapsed_time >= timeout:
                 raise JobTimeoutError(
                     'Timeout while waiting for job {}.'.format(self.job_id))
             time.sleep(wait)
-            status = self.status()
+            stream_status = self._is_streaming()
```

Comment:
I guess we should call self.status() after this while loop is done, so that self._status will have the final status.